### DISCUSSION TITLE
Why are some GraphQL values null?

### DISCUSSION THREAD

**Discussion by scriptsman** [appwrite/appwrite#5292] [Q&A]
### 👟 Reproduction steps

1. Create a document with an attribute type of array, or just watch the $permissions attribute.
2. Query it through Graphql.
3. Get null.
![image](https://user-images.githubusercontent.com/43604482/227756396-fd87c102-d380-4930-9e5d-ec3220fefc5d.png)
4. And it works fine when querying the same document from the web SDK.
![image](https://user-images.githubusercontent.com/43604482/227756433-f7940f6f-ad23-4b28-93e5-17337130dca3.png)

### 👍 Expected behavior

Return the same value as the web SDK.

### 👎 Actual Behavior

Return null.

### 🎲 Appwrite version

Version 1.2.x

### 💻 Operating system

MacOS

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
@scriptsman thanks for creating this issue! 🙏🏼 Would you please try to increase the `_APP_GRAPHQL_MAX_DEPTH` [environment variable](https://appwrite.io/docs/environment-variables#graphql)? Make sure to recreate the docker containers to apply the change.

**Reply by scriptsman** [reply]
I increase the _APP_GRAPHQL_MAX_DEPTH, but get an object string that can not be parsed to an array directly in js.
![image](https://user-images.githubusercontent.com/43604482/228439695-d1a3a78d-8bed-4d45-8991-4ab9e076c59f.png)
Why not return a string that works better with JSON.parse?

**Reply by scriptsman** [reply]
And currently, appwrite GraphQL returns the data as a string. Is it better to return the JSON object?

**Reply by stnguyen90** [reply]
@scriptsman,

> but get an object string that can not be parsed to an array directly in js

I've created an [issue](https://github.com/appwrite/appwrite/issues/5299) for the object. Please 👍🏼 when you get a chance.

> GraphQL returns the data as a string

Yes, this is a limitation of our GraphQL API at the moment. We need to figure out a way to generate the GraphQL schema dynamically and efficiently.
